baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/InvocationTargetExceptionGetCause.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.fixes.SuggestedFixes;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import java.lang.reflect.InvocationTargetException;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = SeverityLevel.SUGGESTION,
+        summary = "Use getCause instead of getTargetException")
+public final class InvocationTargetExceptionGetCause extends BugChecker
+        implements BugChecker.MethodInvocationTreeMatcher {
+
+    private static final Matcher<ExpressionTree> ITE_GET_TARGET_EXCEPTION_MATCHER = Matchers.instanceMethod()
+            .onExactClass(InvocationTargetException.class.getName())
+            .named("getTargetException")
+            .withNoParameters();
+
+    @Override
+    public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+        ExpressionTree caller = ASTHelpers.getReceiver(tree.getMethodSelect());
+        if (caller == null) {
+            return Description.NO_MATCH;
+        }
```

Comment:
I don't think this block is necessary

Reply:
hm ok. This was for:

```java
class Test extends InvocationTargetException {
  @Override
  public Throwable getCause() {
    // maybe some logic
    Throwable t = getTargetException();
    // maybe some logic 
    return t;
  }
}
```

We're saying these are very unlikely to exist, so this isn't needed. will remove.